peekingduck/training/src/data/data_module.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2023 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from albumentations import Compose
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+import pandas as pd
+
+from src.data.base import AbstractDataSet
+from src.data.dataset import PTImageClassificationDataset
+from src.data.data_adapter import DataAdapter
+from src.transforms.augmentations import ImageClassificationTransforms
+from src.utils.general_utils import (
+    create_dataframe_with_image_info,
+    download_to,
+    extract_file,
+    return_list_of_files,
+    stratified_sample_df,
+)
+
+import logging
+from configs import LOGGER_NAME
+
+logger: logging.Logger = logging.getLogger(LOGGER_NAME)  # pylint: disable=invalid-name
+
+
+class ImageClassificationDataModule:
+    """Data module for generic image classification dataset."""
+
+    def __init__(
+        self,
+        cfg: DictConfig,
+        **kwargs: Dict[str, Any],
```

Comment:
`kwargs` seems to be unused in the class, consider either adding `self.kwargs = kwargs` or updating via `self.__dict__.update(kwargs)`.